cpp/src/binary/jit/core/cache.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/error_utils.hpp>
+#include "cache.h"
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+
+namespace cudf {
+namespace jit {
+
+std::string getCacheDir()
```

Comment:
Document this function. Explain the logic used to choose where to cache.

Reply:
done https://github.com/rapidsai/cudf/pull/1245/commits/426e3d89810cd7e69ac9ab21e565575ef0ab4ae3#diff-c9a49624f72ccf3e05fbc58ec2920c5aR38